pageserver/src/layered_repository.rs

```diff
@@ -583,6 +585,13 @@ pub struct LayeredTimeline {
     // because current_logical_size is consistent on last_record_lsn, not ondisk_consistent_lsn
     // NOTE: current_logical_size also includes size of the ancestor
     current_logical_size: AtomicUsize, // bytes
+
+    // needed to exclude calls to .with_label_values and labels formatting to strings
+    // every time we need to change a metric
+    // unfortunately it doesnt forward atomic methods like .fetch_add
+    // so use two fields: actual size and metric
+    // see https://github.com/zenithdb/zenith/issues/622 for discussion
```

Comment:
If we wanted to be fancy, we could implement a custom Metric that reads the value directly from the `current_logical_size` field, when `gather()` is called. Then you wouldn't need to update the gauge separately. Not sure how complicated it would be, or whether it's worth the trouble..

Reply:
I've tried to use metric field directly to store the value, it is possible to implement custom metric which uses desired ordering to manipulate the value. It looks like more natural approach, and I went quite far, but faced some macro trickery and private modules, so decided to stash this and try to workaround it later